### ISSUE TITLE
toString on color created from HSL creates an HSL string, no matter what format is specified

### ISSUE THREAD

**Issue by attaboy** [color-js/color.js#246]
Tested on v0.4.2 and also reproduced on v0.4.0

1. Create a color from an HSL string, e.g. `const color = new Color("hsl(0, 100%, 100%)");`
2. Output the color with toString and specify a format e.g. `color.toString({ format: hex });`

Expected output:
`#FFFFFF` (or `#FFF`)

Actual output:
`hsl(0 100% 100%)`

<img width="1217" alt="Screen Shot 2022-11-21 at 16 45 50" src="https://user-images.githubusercontent.com/71433/203164385-56d0e44d-27e9-414a-a8e2-aa2e3664ec8a.png">

**Comment by LeaVerou** [comment]
This is not a bug per se, as it's working as intended, but there’s definitely a usability issue. 

Named formats are local to color spaces, and hex (in CSS Color) is sRGB-specific. The thinking is that these don't always make sense generically. E.g. what should `new Color("p3", [0.45, 1, 0.3]).toString({format: "hex"})` return? `"#00ff00"` (sRGB coords in hex) or `"#73ff4d"` (P3 coords in hex)?

However, the current behavior is clearly confusing. At the very least Color.js should throw an error that the format was not found instead of just silently discarding it. 

Some other ideas about going further:

- Perhaps it could also walk parent color spaces up if the format was not found in the current color space (converting as necessary), which would fix the hex issue in this case, but things like `new Color("lch", [50, 30, 350]).toString({format: "hex"})` would still not work. We could search all known color spaces, but then what do we do if the same format name exists multiple times?
- Perhaps it should be possible to reference formats from other color spaces by namespacing them (`format: "srgb.hex"` or `format: [srgb, "hex"]`)
- Require formats to be unique globally. I'd rather avoid this.

**Comment by attaboy** [comment]
Ah hah! I guess I have to get out of the "it's sRGB by default" mindset, because I'm used to HSL and sRGB being interchangeable. (And the [docs sort of imply that HSL is based on sRGB](https://colorjs.io/docs/spaces.html#hsl)). I can see why asking for a broader-gamut color to be formatted as hex doesn't make sense of course. I like the idea of walking up the parent color spaces to find an appropriate format, and throwing an error if the format is ambiguous or not found.

In the meantime, `color.to("srgb").toString({ format: "hex" })` does what I want in this situation, and seems strictly better than bothering with a namespaced format.

**Comment by xiegeo** [comment]
I run into the same confusion using this library, trying to get `toString` to work before discovering the `to` method. Now that I re-read the docs just to read docs, I discovered I should have used `display`. When I scanned the document, the parts in the middle are the easiest to miss.

The API is fine, it's a problem with the documentation. Which is hard to get right for a library needing lots of domain knowledge to use. In https://colorjs.io/docs/output.html. That `rgb` is a serialization format while `srgb` is a color space is one of many confusions for beginners like myself. My suggestion is to add a tldr at the beginning to use `display` as a cross-browser method to set color, instead of building up to it and having readers get lost or start experimenting with less idea solutions.
